examples/cli/cli.cpp

```diff
@@ -1066,8 +1067,45 @@ int main(int argc, char ** argv) {
     }
 
     for (int f = 0; f < (int) params.fname_inp.size(); ++f) {
-        const auto fname_inp = params.fname_inp[f];
-		const auto fname_out = f < (int) params.fname_out.size() && !params.fname_out[f].empty() ? params.fname_out[f] : params.fname_inp[f];
+        const auto & fname_inp = params.fname_inp[f];
+        class FnameOutFactory {
+            std::string buf;
+            const size_t basename_length;
+            bool used_stdout;
+
+        public:
+            const bool is_stdout;
+            decltype(whisper_print_segment_callback) * const print_segment_callback;
+
+            FnameOutFactory(const std::string & fname_out, const std::string & fname_inp, whisper_params & params) :
+                    buf{!fname_out.empty() ? fname_out : fname_inp},
+                    basename_length{buf.size()},
+                    is_stdout{buf == "-"}, used_stdout{params.diarize},
```

Comment:
Perhaps place `used_stdout` on a separate line for consistency with the other fields.